src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,198 @@
+package duke.command;
+import duke.dukeException.DukeException;
+import duke.task.*;
+import java.io.*;
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public class Command {
+    /** The input command by user */
+    public String commandName;
+
+    /**
+     * Class constructor.
+     */
+    public Command(String commandName) {
+        this.commandName = commandName;
+    }
+
+
+    /**
+     * Processes tasks according to the command.
+     *
+     * @param tasks  the tasks before execution.
+     * @throws IOException  If an input or output
+     *                      exception occurred
+     */
+    public void execute(TaskList tasks) throws IOException, DukeException {
+        ArrayList<Task> tList = tasks.tasks;
+        if (commandName.equals("list")) {
+            System.out.println("     Here are the tasks in your list:");
+            int counter = 1;
+            for(Task s : tList) {
+                if (s != null) {
+                    System.out.println("     " + counter + "." + s.toString());
+                    counter++;
+                } else {
+                    break;
+                }
+            }
+        } else if (commandName.split(" ")[0].equals("done")) {
+            int index = Integer.parseInt(commandName.split(" ")[1]);
+            tList.get(index-1).markAsDone();
+            System.out.println("     Nice! I've marked this task as done:");
+            System.out.println("       " + tList.get(index-1).toString());
+            Task currTask = tList.get(index-1);
+            tasks.tasks = tList;
+            String old = null;
```

Comment:
Perhaps this name could be puzzling as readers might wonder what "old" means? Maybe a clearer name would be better.